Source/JavaScriptCore/jit/JITThunks.cpp

```diff
@@ -207,12 +219,12 @@ void JITThunks::finalize(Handle<Unknown> handle, void*)
     }
 }
 
-NativeExecutable* JITThunks::hostFunctionStub(VM& vm, TaggedNativeFunction function, TaggedNativeFunction constructor, ImplementationVisibility implementationVisibility, const String& name)
+NativeExecutable* JITThunks::hostFunctionStub(VM& vm, JSGlobalObject* globalObject, TaggedNativeFunction function, TaggedNativeFunction constructor, ImplementationVisibility implementationVisibility, const String& name)
 {
-    return hostFunctionStub(vm, function, constructor, nullptr, implementationVisibility, NoIntrinsic, nullptr, name);
+    return hostFunctionStub(vm, globalObject, function, constructor, nullptr, implementationVisibility, NoIntrinsic, nullptr, name);
 }
 
-NativeExecutable* JITThunks::hostFunctionStub(VM& vm, TaggedNativeFunction function, TaggedNativeFunction constructor, ThunkGenerator generator, ImplementationVisibility implementationVisibility, Intrinsic intrinsic, const DOMJIT::Signature* signature, const String& name)
+NativeExecutable* JITThunks::hostFunctionStub(VM& vm, JSGlobalObject* globalObject, TaggedNativeFunction function, TaggedNativeFunction constructor, ThunkGenerator generator, ImplementationVisibility implementationVisibility, Intrinsic intrinsic, const DOMJIT::Signature* signature, const String& name)
```

Comment:
JIT stub is shared in one VM with multiple JSGlobalObject so we cannot have JSGlobalObject.

Reply:
the `JSGlobalObject` is not actually saved in the `NativeExecutable`.  it's just used inside `NativeExecutable::create` to get any related `Debugger`

is the issue just passing it at all, or specifically if it's saved as a member variable?

Reply:
Yes, passing `JSGlobalObject*` is an issue.